### PR TITLE
Create scripts to facilitate X.509 auth testing for standalone Edge Hub and downstream devices

### PR DESCRIPTION
**Pull request by mrohera**

### CODE REVIEW

File: tools/CACertificates/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Managing test CA certificates for samples and tutorials
+
+## WARNING
```
Comment (myagley) [inline tools/CACertificates/README.md]:
```suggestion
## :warning: WARNING :warning:
```

File: tools/CACertificates/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+Certificates created by these scripts **MUST NOT** be used for production.  They contain hard-coded passwords ("1234"), expire by default after 30 days, and most importantly are provided for demonstration purposes to help you quickly understand CA Certificates.  When productizing against CA Certificates, you'll need to use your own security best practices for certification creation and lifetime management.
+
+## Introduction
+This document helps create certificates for use in **pre-testing** IoT SDK's against the IoT Hub and Edge runtime.  In particular, the tools in this directory can be used to either setup CA Certificates (along with proof of possession) or Edge device certificates.  This document assumes you have basic familiarity with the scenario you are setting up for as well as some knowledge of PowerShell or Bash.
```
Comment (myagley) [inline tools/CACertificates/README.md]:
```suggestion
This document helps create certificates for use in **pre-testing** IoT SDKs against the IoT Hub and Edge runtime.  In particular, the tools in this directory can be used to either setup CA Certificates (along with proof of possession) or Edge device certificates.  This document assumes you have basic familiarity with the scenario you are setting up for as well as some knowledge of PowerShell or Bash.
```

File: tools/CACertificates/README.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+* `cp *.cnf` and `cp ca-certs.ps1` from the directory this .MD file is located into your working directory.
+* Run `Set-ExecutionPolicy -ExecutionPolicy Unrestricted`.  You'll need this for PowerShell to allow you to run the scripts.
+* Run `. .\ca-certs.ps1` .  This is called dot-sourcing and brings the functions of the script into PowerShell's global namespace.
+* Run `Test-CACertsPrerequisites` to make sure all the required pre requisites are met and follow on the on screen prompts to resolve.
+
+###  **Bash**
+* Start Bash.
+* `cd` to the directory you want to run in.  All files will be created as children of this directory.
+* `cp *.cnf` and `cp certGen.sh` from the directory this .MD file is located into your working directory.
+* `chmod +x certGen.sh`
```
Comment (myagley) [inline tools/CACertificates/README.md]: You should be able to check this file in with execute mode. Then this step wouldn't be needed on linux.

File: tools/CACertificates/certGen.sh [hunk 4]
```diff
@@ -0,0 +1,407 @@
+## Copyright (c) Microsoft. All rights reserved.
```
Comment (myagley) [inline tools/CACertificates/certGen.sh]: This is a total nit, but is it possible to name this script in kabob-case like the powershell script?

Comment (mrohera) [reply tools/CACertificates/certGen.sh]: Several pieces of our MS documentation use this script file after syncing this out from the C SDK repo. Believe me this naming convention does not work for me as well :) but I am not sure I can change the name.

Comment (myagley) [reply tools/CACertificates/certGen.sh]: Ah. Gotcha. Ok, we should keep it.

**Pull request merged**